pkg/download/classic/downloader_test.go

```diff
@@ -297,15 +296,6 @@ func TestDownload_FilteringCanBeTurnedOffViaFeatureFlags(t *testing.T) {
 	assert.Len(t, configurations, 1)
 }
 
-func TestDownload_EmptyAPIMap_ResultsInError(t *testing.T) {
```

Comment:
As the behavior has changed and the classic downloader silently returns an empty list of configurations, we're not telling users anymore if there where really no configs on the environment, or filtering has somehow removed all APIs and nothing was downloaded in the first place. 

I think we should still be clear about that. 
Please either reintroduce this check and error return in Download or (if it's possible on that level) differentiate the cases in download_configs.go and log differently for each case. 

Related to this: Please change the signature or `Download` to how it actually functions. As it does not accept API to filter for anymore, please remove that function argument. Also depending on the changes you decide for this case, remove the error return. It does not make sense to have that in the signature if the method always returns a nil error

Reply:
Currently, during the process of evaluation of input data for every unknown classic API endpoint will be reported as an error to the user (and the process will be terminated). So the user will only be able to continue the process if he enters the correct name.
By the point where `downloaders.Classic().Download(opts.projectName)` is called, all input data is clean, and all irregularities should be reported to the user.
_Is it enough to log (info level) the number of endpoints from which config is going to be downloaded?_

Signature of Download function is: `func (d *Downloader) Download(projectName string, _ ...config.ClassicApiType) (project.ConfigsPerType, error)`
To implement `type Downloader[T v2.Type] interface` second argument needs to be present.

Reply:
Thanks, missed the generic interface part.

As discussed directly: please add logging so a user can still get that nothing is being downloaded without debug logging - under that assumption that in normal operation/usage of the interface this can never happen, no error is needed for this case.